packages/contracts/test/dollar/core/StakingShare.t.sol

```diff
@@ -383,4 +388,29 @@ contract StakingShareTest is DepositStakingShare {
         vm.prank(fifthAccount);
         stakingShare.setUri(1, stringTest);
     }
+
+    function testUUPS_ShouldUpgradeAndCall() external {
+        StakingShareOneChange stakingShareOneChange = new StakingShareOneChange();
+
+        string
+            memory uri = "https://bafybeifibz4fhk4yag5reupmgh5cdbm2oladke4zfd7ldyw7avgipocpmy.ipfs.infura-ipfs.io/";
+
+        vm.startPrank(admin);
+        bytes memory hasUpgradedCall = abi.encodeWithSignature("hasUpgraded()");
+
+        (bool success, ) = address(IStakingShareToken).call(hasUpgradedCall);
+        assertEq(success, false, "should not have upgraded yet");
+        require(success == false, "should not have upgraded yet");
+
+        IStakingShareToken.upgradeTo(address(stakingShareOneChange));
+
+        (success, ) = address(IStakingShareToken).call(hasUpgradedCall);
+        assertEq(success, true, "should have upgraded");
+        require(success == true, "should have upgraded");
+
+        vm.expectRevert();
+        IStakingShareToken.initialize(address(diamond), uri);
```

Comment:
You must be doing something wrong here, why does it initializes and `diamond` address, when core are separate from diamonds, should have it's own address

Reply:
If you mean why am I initialising using the diamond address it's because Diamond is the ManagerFacet which is the storage location for the UUPS proxy addresses. It doesn't need to have a separate contract for that? Diamond can perform as the manager as it already is simply by casting the functions we need from it like the managerFacet and accessControl.

In the above test it does not re-initialize as expected, sorry but I'm unsure why you think something is wrong?